pythonlib/src/mill/pythonlib/PythonModule.scala

```diff
@@ -21,59 +40,81 @@ trait PythonModule extends Module {
     PathRef(python)
   }
 
-  def typeCheck: T[Unit] = Task {
-    Task.traverse(moduleDeps)(_.typeCheck)()
-
-    os.call(
-      (pythonExe().path, "-m", "mypy", "--strict", sources().path),
-      stdout = os.Inherit,
-      cwd = T.workspace
+  // TODO: right now, any task that calls this helper will have its own python
+  // cache. This is slow. Look into sharing the cache between tasks.
+  protected def runPythonExe(args: Task[Seq[String]]): Task[Unit] = Task.Anon {
```

Comment:
Rather than a method taking and returning `Task`s, we should have a zero-arg method returning a `Task` that returns a value that has more methods we can call. See how this is done for `runner().run(...)` in `JavaModule`s and `ScalaModule`s